packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+        // production application
+        const testDataObject2 = await requestFluidObject<TestDataObject>(summarizingContainer2, testDataObjectId);
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer2);
+
+        // The request pattern should fail!
+        assert.throws(() => testDataObject2._root.set("send", "op"),
+            (error) => {
+                const correctErrorType = error.errorType = "dataCorruptionError";
+                const correctErrorMessage = error.errorMessage?.startsWith(`Context was tombstoned`) === true;
+                return correctErrorType && correctErrorMessage;
+            },
+            `Should not be able to send ops for a tombstoned datastore.`);
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sessionExpiryTimeoutMs` above
+    it("Container loaded before sweep timeout expires can't send ops for tombstoned datastores", async () => {
+        const { testDataObjectId, summaryVersion } = await createUnreferencedDataStore();
+
+        // Wait some time, the datastore should not be sweep ready after this wait
+        await delay(sweepTimeoutMs - wait);
+
+        // Load a new container and summarizer based on the latest summary, summarize
+        const {
+            container: summarizingContainer2,
+            summarizer: summarizer2,
+        } = await loadSummarizerAndContainer(summaryVersion);
+
+        // Use the request pattern to get the testDataObject - this is unsafe and no one should do this in their
+        // production application.
+        const testDataObject2 = await requestFluidObject<TestDataObject>(summarizingContainer2, testDataObjectId);
+
+        // Wait enough time so that the datastore is sweep ready
+        await delay(wait);
+
+        // Send an op to update the currentTimestampMs to now
+        const mainDataStore2 = await requestFluidObject<TestDataObject>(summarizingContainer2, "default");
+        mainDataStore2._root.set("send a", "op");
+
+        // The datastore should be tombstoned now
+        await summarize(summarizer2);
+
+        // The request pattern should fail!
+        assert.throws(() => testDataObject2._root.set("send", "op"),
+            (error) => {
+                const correctErrorType = error.errorType = "dataCorruptionError";
+                const correctErrorMessage = error.errorMessage?.startsWith(`Context was tombstoned`) === true;
+                return correctErrorType && correctErrorMessage;
+            },
+            `Should not be able to send ops for a tombstoned datastore.`);
+    });
+
+    // If this test starts failing due to runtime is closed errors try first adjusting `sessionExpiryTimeoutMs` above
+    itExpects("Container loaded after sweep timeout expires can't receive ops for tombstoned datastores",
+    [
+        {
+            eventName: "fluid:telemetry:ContainerRuntime:GarbageCollector:SweepReadyObject_Loaded",
```

Comment:
How come these errors are not fired for the previous 2 tests?

Reply:
The test data object is loaded before the summary marks them as sweep ready. They're considered unreferenced, hmmm maybe there should be some inactiveObjectX telemetry

Reply:
So I think what's happening is that inactiveObjectX telemetry isn't fired because the summary hadn't yet marked them as inactive as GC hadn't run. Will investigate further

Reply:
After an investigation we had determined that only summarizer clients hadn't loaded their initial base gc state which meant that loading a datastore that should be sweep ready would not fire as the gc state was not initialized yet. We are putting a fix in to load the gc state for summary clients as well once they are caught up and set to active.